ARCs/arc-0063.md

```diff
@@ -0,0 +1,159 @@
+
+    ```python
+    def generate_32bytes_from_addresses(addr1, addr2, sk):
+    combined = addr1 + addr2
+    combined_signed = util.sign_bytes(combined.encode(), sk)
+    hash_digest = hashlib.sha256(combined_signed.encode()).digest()
+    seed = hash_digest[:32]
+    sk = SigningKey(seed,encoder=nacl.encoding.RawEncoder)
+    vk = sk.verify_key
+    a = encoding.encode_address(vk.encode())
+    return  base64.b64encode(sk.encode() + vk.encode()).decode(), a
+
+    plug_in_sk, plug_in_addr  = generate_32bytes_from_addresses(bob_addr, lsig.address(), bob_sk)
+    ```
+
+2. **Sign Lsig with Plug-In Signer**:
+    - Sign the Lsig using the plug-in signer.
+    - Publish the public signature on the indexer.
+
+    ```python
+        public_key, secret_key = nacl.bindings.crypto_sign_seed_keypair(base64.b64decode(plug_in_sk)[: constants.key_len_bytes])
+        message = constants.logic_prefix + program
+        raw_signed = nacl.bindings.crypto_sign(message, secret_key)
+        crypto_sign_BYTES = nacl.bindings.crypto_sign_BYTES
+        signature = nacl.encoding.RawEncoder.encode(raw_signed[:crypto_sign_BYTES])
+        plug_in_public_sig = base64.b64encode(signature).decode()
+    ```
+
+3. **Create 1/3 Msig Account**:
+    - Create a multi-signature account with owner address, the Lsig address, and the plug-in signer.
+
+    ```python
+    bob_vault_msig = transaction.Multisig(1,1,[bob_addr, lsig.address(), plug_in_addr])
+    ```
+
+    - Add a transaction note to the transaction to help third party to retrieve signer and vault information.
+
+    ```json
+        {
+            "s": bob_addr,
+            "lsigs": lsig.address(),
+            "sigA": plug_in_addr,
+            "sigS": plug_in_public_sig,
+            "vault": bob_vault_msig.address(),
+        }
+    ```
+
+    - Prefix the note following the [ARC-2](./arc-0002.md) standard. `arc_63:j:`
+
+    ```python
+    ptxn = transaction.PaymentTxn(
+        bob_addr, sp, bob_vault_msig.address(), int(1e6), note=f"arc_63:j:{note_field}"
+    ).sign(bob_sk)
+    ```
+
+4. **Opt-In to Msig Vault**:
+    - Anyone can opt-in to the Msig vault using the plug-in signer’s public address and the published signature.
+
+    ```python
+    optin_txn = AssetTransferTxn(
+        sender=bob_vault_msig.address(),
+        sp=sp,
+        receiver=bob_vault_msig.address(),
+        amt=0,
+        index=a_id,
+    )
+    lsig.lsig.msig = bob_vault_msig
+    lsig.lsig.msig.subsigs[2].signature = base64.b64decode(plug_in_public_sig) # signature from plug_in_public
+    lstx = LogicSigTransaction(optin_txn, lsig)
+    ```
+
+5. **Rekey Plug-In Signer**:
+    - Rekey the plug-in signer to the zero address to prevent any further usage.
+
+### Rekeying Process
+
+The plug-in signer is rekeyed to the zero address to eliminate the risk of unauthorized transactions. This ensures that the signer cannot be used post-creation, maintaining the integrity of the multi-signature setup.
```

Comment:
It's not possible to rekey to the zero address, as that would result in a blank field and being omitted from the transaction. I would suggest rekeying to the owner address instead... but now I'm wondering if rekeyed accounts even allow for delegate signatures to work. It may invalidate the signature. Will test and report back.

Reply:
I wanted to do something like this :
```python
zero_msig = transaction.Multisig(1,1,[constants.ZERO_ADDRESS])
rekey_txn = transaction.PaymentTxn(
    plug_in_addr, sp, plug_in_addr, 0, rekey_to=zero_msig.address()
)
```

Reply:
Just to loop back on this. Sure enough if you rekey the account, the existing delegate signature is now no longer valid.
1. AccountA signs an Lsig to create a delegate signature
2. AccountA rekeys to AccountB
3. A txn is sent using AccountA's delegate signature - FAIL `should have been authorized by AccountB but was actually authorized by AccountA`